### ISSUE TITLE
False positive `literal_string_with_formatting_args` for non-static templete string expansion with replace.

### ISSUE THREAD

**Issue by anatawa12** [rust-lang/rust-clippy#13989]
### Summary

When I write code expanding dynamically changed template string, I encountered `literal_string_with_formatting_args`.

I think calling `replace("{var}", &var)` is likely to be intended to not treat as format string in rust.

### Lint Name

literal_string_with_formatting_args

### Reproducer

I tried this code:

```rust
fn lint_test(templete: &str) -> String {
    let version = env!("CARGO_PKG_VERSION");
    templete.replace("{version}", version)
}
```

I saw this happen:

```
warning: this looks like a formatting argument but it is not part of a formatting macro
 --> vrc-get-gui/src/main.rs:3:23
  |
3 |     templete.replace("{version}", version)
  |                       ^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#literal_string_with_formatting_args
  = note: `#[warn(clippy::literal_string_with_formatting_args)]` on by default
```

I expected to see this happen:
No lint for this code.

### Version

```text
rustc 1.85.0-beta.1 (e30eefff4 2025-01-08)
binary: rustc
commit-hash: e30eefff41038ceea427009023627d6d66b36715
commit-date: 2025-01-08
host: aarch64-apple-darwin
release: 1.85.0-beta.1
LLVM version: 19.1.6
```

### Additional Labels

the commit message and description of [PR adds `literal_string_with_formatting_args`](https://github.com/rust-lang/rust-clippy/pull/13410) says `literal_string_with_formatting_args` is `pedantic` so it might be it's a bug that `literal_string_with_formatting_args` is categorized as `suspicious`;

According to the edit history of the PR description, it's originally categorized as `suspicious` and changed to `pedantic` later, but code has not changed (or rebase miss in some point).

When I [searched](https://github.com/search?q=lang%3ARust+%2Freplace%5C%28%22%5C%7B%28%5Cw%2B%29%5C%7D%22%2C+%2F&type=code) on github, I can see the thousands of similar false positive cases are exists.

**Comment by Xuanwo** [comment]
Oops, during my in-depth research, I found that this issue is a duplicate of https://github.com/rust-lang/rust-clippy/issues/13989. It is also likely to be resolved in https://github.com/rust-lang/rust-clippy/pull/13953.

I believe it's safe to close this issue now.

**Comment by anatawa12** [comment]
I already saw #13953 but I think this is not fixed by #13953 since #13953 is about proc macro or something like this, but this is about real, hand-written code.
